app/javascript/controllers/tag_select_controller.js

```diff
@@ -35,9 +35,14 @@ export default class extends Controller {
 
   change() {
     const valueWas = this.element.dataset.valueWas
-    const values = Array.from(this.element.selectedOptions).map((o) => o.value)
+    const initialValues = Array.from(this.element.selectedOptions).map((o) => o.value)
+    const values = this.dedupeValues(initialValues)
     const valueIs = JSON.stringify(values)
 
+    if (values.length !== this.element.selectedOptions.length) {
+      Tags.getInstance(this.element).removeLastItem()
```

Comment:
the only thing i'm unsure about this is that this assumes that the user is typing in categories one at a time. i'm not sure if there are other ways the user has of bulk "importing" categories or entering a bunch of different selections in one swoop? this method just removes the last thing the user tries to input, if it was a dupe.

Reply:
Tried a few things (pasting in multiple lines, tabs, etc)... couldn't get it to add more than one tag at a time.  So this seems safe?

Perusing the docs a bit - there is also a [onCanAdd](https://github.com/lekoala/bootstrap5-tags?tab=readme-ov-file#options) event that would accomplish this (returning false means you can't add the thing).  But this works just as well.